### PR TITLE
Implement RustjitCodegen::compile

### PR DESCRIPTION
**Pull request by Riey**

### CODE REVIEW

File: codegen/rustjit.rs [hunk 1]
```diff
@@ -17,16 +17,30 @@ pub struct RustjitContext {
...
-            IrValue::Variable(ty, id) => todo!(),
-            IrValue::Register(ty, id) => todo!(),
-            IrValue::Constant(constant) => todo!(),
+            IrValue::Variable(_, id) => {
+                self.variables.insert(id, data);
+            }
+            IrValue::Register(_, id) => {
+                self.registers.write(id, &data.to_le_bytes());
+            }
+            IrValue::Constant(_) => panic!("Contant cannot be set"),
```
Comment (ArtBlnd) [inline codegen/rustjit.rs]: Should we only accept `Variable` or `Register`  here?

File: codegen/rustjit.rs [hunk 2]
```diff
@@ -17,16 +17,30 @@ pub struct RustjitContext {
...
         match value {
-            IrValue::Variable(ty, id) => todo!(),
-            IrValue::Register(ty, id) => todo!(),
-            IrValue::Constant(constant) => todo!(),
+            IrValue::Variable(_, id) => {
+                self.variables.insert(id, data);
+            }
+            IrValue::Register(_, id) => {
+                self.registers.write(id, &data.to_le_bytes());
+            }
```
Comment (ArtBlnd) [inline codegen/rustjit.rs]: New value should not overwrite old value if the size of new value is smaller than old value. 
e.g, assigning value `0xFFFF.ty() == U16` to `0xFFFF0000.ty() == U32`.  should be `0xFFFFFFFF`

File: codegen/rustjit.rs [hunk 3]
```diff
@@ -17,16 +17,30 @@ pub struct RustjitContext {
 impl RustjitContext {
     pub fn get(&self, value: IrValue) -> u64 {
```
Comment (ArtBlnd) [inline codegen/rustjit.rs]: I think it is better introduce a intermediate(name example: `RustjitIV`) value than using u64 here.

Comment (Riey) [reply codegen/rustjit.rs]: I'll try

File: codegen/rustjit.rs (hunk 1 above)
Comment (Riey) [reply codegen/rustjit.rs]: Yes, I think set `Constant` doesn't make sense.

**Pull request closed without merging**